src/hooks/use-context-menu.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useCallback, useState } from 'preact/hooks';
+
+const useContextMenu = (outerRef: any) => {
+  const [xPos, setXPos] = useState('0px');
+  const [yPos, setYPos] = useState('0px');
+  const [menu, showMenu] = useState(false);
```

Comment:
I think this pair should be renamed as `[isMenuVisible, setIsMenuVisible]`, owing to the useState conventions